packages/backend/src/api/job/jobRepository.ts

```diff
@@ -0,0 +1,166 @@
+
+import { db } from '../../db';
+import { Job, JobCategory, JobCategorySchema, JobRequest, JobSavedSchema, JobSchema, SavedJob } from '../job/jobModel';
+//import { JobRequest } from './jobRequests';
+
+export const jobRepository = {
+  findJobsAsync: async (conditions: SQL[], limit: number, offset: number): Promise<Job[]> => {
+    let baseQuery = db
+      .select()
+      .from(job)
+      .leftJoin(jobCategory, eq(job.categoryId, jobCategory.id))
+      .leftJoin(jobLevel, eq(job.levelId, jobLevel.id))
+      .leftJoin(company, eq(job.companyId, company.id));
+
+    if (conditions.length > 0) {
+      baseQuery = baseQuery.where(and(...conditions)) as typeof baseQuery;
+    }
+    const result = await baseQuery.limit(limit).offset(offset).execute();
+    const parsedJobs = result.map((row) => {
+      const jobData = {
+        ...row.job,
+        categoryTitle: row.job_category?.title,
+        levelTitle: row.job_level?.title,
+        companyName: row.company?.name,
+      };
+      const parsed = JobSchema.parse(jobData);
+      return parsed;
+    });
+
+    return parsedJobs;
+  },
+
+  findJobsCountAsync: async (conditions: SQL[]): Promise<number> => {
+    const countQuery = db
+      .select({ count: sql<number>`count(*)` })
+      .from(job)
+      .leftJoin(jobCategory, eq(job.categoryId, jobCategory.id))
+      .leftJoin(jobLevel, eq(job.levelId, jobLevel.id))
+      .leftJoin(company, eq(job.companyId, company.id));
+
+    if (conditions.length > 0) {
+      countQuery.where(and(...conditions));
+    }
+
+    const result = await countQuery.execute();
+    return result[0]?.count ?? 0;
+  },
+
+  findJobByIdAsync: async (id: string): Promise<Job | null> => {
+    const result = await db
+      .select()
+      .from(job)
+      .leftJoin(jobCategory, eq(job.categoryId, jobCategory.id))
+      .leftJoin(jobLevel, eq(job.levelId, jobLevel.id))
+      .leftJoin(company, eq(job.companyId, company.id))
+      .where(eq(job.id, id))
+      .limit(1);
+    if (result.length === 0) return null;
+    const row = result[0];
+    const jobData = {
+      ...row.job,
+      categoryTitle: row.job_category?.title,
+      levelTitle: row.job_level?.title,
+      companyName: row.company?.name,
+    };
+    const parsedJob = JobSchema.parse(jobData) || null;
+    return parsedJob;
+  },
+
+  deleteJobByIdAsync: async (id: string): Promise<Job | null> => {
+    const result = await db.delete(job).where(eq(job.id, id));
+    if (result.length === 0) return null;
+    return JobSchema.parse(result[0]) || null;
+  },
+
+  updateJobAsync: async (id: string, updateJobRequest: JobRequest): Promise<Job> => {
+    console.log('update');
+
+    const updateJob = await db
+      .update(job)
+      .set({
+        ...updateJobRequest,
+        updatedAt: new Date(),
+      })
+      .where(eq(job.id, id))
+      .returning();
+    return JobSchema.parse(updateJob[0]);
```

Comment:
could this be a nullpointerexception ? updateJob[0]

Reply:
no because parse we are updating and if there are somthng wrong with the input the parse will throw an error so the catch of in the service will handle it